crates/proto/src/quic/quic_client_stream.rs

```diff
@@ -270,40 +285,45 @@ impl QuicClientStreamBuilder {
 }
 
 /// Default crypto options for quic
-pub fn client_config_tls13_webpki_roots() -> TlsClientConfig {
-    use rustls::{OwnedTrustAnchor, RootCertStore};
+pub fn client_config_tls13() -> Result<TlsClientConfig, ProtoError> {
+    use rustls::RootCertStore;
+    #[cfg_attr(
+        not(any(feature = "native-certs", notfeature = "webpki-roots")),
+        allow(unused_mut)
+    )]
     let mut root_store = RootCertStore::empty();
+    #[cfg(all(feature = "native-certs", not(feature = "webpki-roots")))]
+    {
+        use crate::error::ProtoErrorKind;
+
+        for cert in rustls_native_certs::load_native_certs()? {
+            if let Err(err) = root_store.add(&rustls::Certificate(cert.0)) {
+                tracing::warn!(
+                    "failed to parse certificate from native root store: {:?}",
+                    &err
+                );
+            }
+        }
+        if root_store.is_empty() {
+            return Err(ProtoErrorKind::NativeCerts.into());
+        }
```

Comment:
The thing I did here in general was to log warnings when encountering invalid certificates and returning an error if either `load_native_certs()` failed or if no certs were found at all.

In comparison to how `reqwest` does it, they also error out if `load_native_certs()` fails but they don't error out if no certificates at all where found, only if at least one invalid and no valid certificate was found.

See https://github.com/seanmonstar/reqwest/blob/06c8e5b0b008afee8114fb979b85cd8b73415391/src/async_impl/client.rs#L381-L425.